rest/blip_client_test.go

```diff
@@ -716,3 +728,32 @@ func (btr *BlipTesterReplicator) storeMessage(msg *blip.Message) {
 	defer btr.messagesLock.Unlock()
 	btr.messages[msg.SerialNumber()] = msg
 }
+
+func (btc *BlipTesterClient) WaitForBlipMessage(docId, revId string) (msg *blip.Message, found bool) {
+	ticker := time.NewTicker(50 * time.Millisecond)
+	timeout := time.After(10 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			btc.rt.tb.Fatalf("BlipTesterClient timed out waiting for BLIP message docId: %v, revId: %v", docId, revId)
+			return nil, false
+		case <-ticker.C:
+			if data, found := btc.GetBlipMessage(docId, revId); found {
+				return data, found
+			}
+		}
+	}
+}
+
+func (btc *BlipTesterClient) GetBlipMessage(docId, revId string) (msg *blip.Message, found bool) {
+	btc.docsLock.RLock()
+	defer btc.docsLock.RUnlock()
+
+	if rev, ok := btc.docs[docId]; ok {
+		if pair, found := rev[revId]; found {
+			return pair.message, found
```

Comment:
For a DocID/RevID created on the client, `pair.message` is going to be nil, but we're going to return a `found=ok` bool, because the rev exists.

Is the expectation that tests need to assert NotNil on the message first? or should we change the `found` value to correspond to whether the given DocID/RevID has an associated BLIP message?

Reply:
Yeah, it would be better to change the found value to correspond to whether the given DocID/RevID has an associated BLIP message. Fixed in latest revision.